packages/cli-upload/src/upload.js

```diff
@@ -87,15 +96,28 @@ export const upload = command('upload', {
       let { dir, name, ext } = path.parse(relativePath);
       img.type = ext === '.png' ? 'png' : 'jpeg';
       img.name = path.join(dir, name);
+      let snapshotName = config.stripExtensions ? img.name : relativePath;
 
-      percy.upload({
-        name: config.stripExtensions ? img.name : relativePath,
-        // width and height is clamped to API min and max
-        widths: [Math.max(10, Math.min(img.width, 2000))],
-        minHeight: Math.max(10, Math.min(img.height, 2000)),
-        // resources are read from the filesystem as needed
-        resources: () => getImageResources(img)
-      });
+      if (tokenType === 'web') {
+        percy.upload({
+          name: snapshotName,
+          // width and height is clamped to API min and max
+          widths: [Math.max(10, Math.min(img.width, 2000))],
+          minHeight: Math.max(10, Math.min(img.height, 2000)),
+          // resources are read from the filesystem as needed
+          resources: () => getImageResources(img)
+        });
+      } else if (tokenType === 'generic') {
+        percy.upload({
+          name: snapshotName,
+          tag: BYOS_TAG,
+          tiles: [
+            { filepath: img.absolutePath }
+          ]
+        });
+      } else {
+        throw new Error('Invalid Token Type');
```

Comment:
```suggestion
        throw new Error('Invalid Token Type, only web, self managed token type are allowed.');
```

lets ask internally about the wordings here.